faq/slurm.inc

```diff
@@ -91,15 +93,17 @@ href=\"?category=openfabrics#ib-locked-pages-more\">this FAQ entry for
 references about Slurm</a>.";
 
 /////////////////////////////////////////////////////////////////////////
+$q[] = "My job fails when using mpirun under Slurm 20.11";
 
-$q[] = "Any issues with Slurm 2.6.3?";
+$anchor[] = "slurm-20.11-mpirun";
 
-$anchor[] = "slurm-2.6.3-issue";
-
-$a[] = "Yes. The Slurm 2.6.3, 14.03 releases have a bug in their PMI-2
-support.
-
-For the slurm-2.6 branch, it is recommended to use the latest version
-(2.6.9 as of 2014/4), which is known to work properly with pmi2.
-
-For the slurm-14.03 branch, the fix will be in 14.03.1.";
+$a[] = "Due to the change in Slurm 20.11 - steps being exclusive by default
+using mpirun when build --with-slurm in sbatch script or under salloc may
+result in orted started process being limited to subset of resources. If you
```

Comment:
The wording here is very confusing...

"`the change`" => What change? Can a link to the Slurm 20.11 changelog be included here?

"`when build --with-slurm`" => "`when Open MPI is configured with [--with-slurm]`"?

"`in sbatch script`" => "`in a job script submitted via [sbatch]`"?

"`orted started process`" => "`processes started via orted`"?

etc.

How would somebody notice that a subset of resources is being used? Also, how bad of a subset, how much potential impact is there on performance?
Is there any easy way to check?

FInally: a lot of HPC sites are going to be taken by surprise by this, may not notice at all (or too late), or may lose a lot of time figuring out why performance of MPI applications has suddenly taken a deep dive. This sort of thing is not exactly easy to search for, and it's mildly infuriating (to put it mildly) that an environment variable like `$SLURM_WHOLE` (please include a pointer to the Slurm documentation that explains what this does...) magically solves the problem.

This should be considered a regression in Slurm. I'm happy to open a ticket with Slurm support on this to follow up.

Reply:
we were caught off-guard by this change (Compute Canada, Cedar cluster). `mpirun` can actually work but since on higher nodes it will put all processes on core 0 (with cgroups in Slurm enabled at least), the performance is then devastating:

this is HPL on a tiny matrix by today's standards on 2 nodes with 32 cores each, using plain `srun` and plain `mpirun`:
```
T/V                N    NB     P     Q               Time                 Gflops
---------------------------------------------------------------------------------
srun:
WR11C2R4        6000   192     8     8               0.22              6.490e+02
mpirun:
WR11C2R4        6000   192     8     8            1170.97              1.230e-01
```